cpu/nrf52/include/periph_cpu.h

```diff
@@ -170,9 +170,9 @@ typedef enum {
  * Each device supports up to 4 channels. If you want to use less than 4
  * channels, just set the unused pins to GPIO_UNDEF.
  *
- * @note    define unused pins only from right to left, so the defined channels
- *          always start with channel 0 to x and the undefined ones are from x+1
- *          to PWM_CHANNELS.
```

Comment:
This is still valid from an API perspective.
`tests/periph_pwm` iterates from 0 to `pwm_channels()`, so if there were gaps, it would miss some channels.

(And since you can pretty much map any pin to any function on nrf52, there is really no reason for gaps)

Reply:
That test justifies leaving it in at least for the scope of this (still adding the warning as well).

I see some further need for cleanup here (differentiating in the documentation between generic rules and per-cpu peculiarities, currently [the pwm_conf_t](http://riot-os.org/api/structpwm__conf__t.html) are a conglomerate of both)) and would also consider this left-to-right filling for reevaluation (for the EFM32 doesn't have such flexible routing, so rather than iterating from channel 0 to pwm_channels(dev) it may be better to iterate from 0 to PWM_CHANNELS and only act if pwm_is_active(dev, channel)), but that's not for now.